src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -16,18 +19,26 @@ public static class WebHostExtensions
         /// <param name="host">The <see cref="IWebHost"/> to run.</param>
         public static void Run(this IWebHost host)
         {
+            AutoResetEvent done = new AutoResetEvent(false);
             using (var cts = new CancellationTokenSource())
             {
+#if NETSTANDARD1_5
+                var applicationLifetime = host.Services.GetService<IApplicationLifetime>();
+                AssemblyLoadContext.Default.Unloading += context => {
+                    Shutdown(cts);
+
+                    done.WaitOne();
+                };
+#endif
                 Console.CancelKeyPress += (sender, eventArgs) =>
                 {
-                    Console.WriteLine("Application is shutting down...");
-                    cts.Cancel();
+                    Shutdown(cts);
 
-                    // Don't terminate the process immediately, wait for the Main thread to exit gracefully.
-                    eventArgs.Cancel = true;
```

Comment:
How big of a change is this from the default behavior? Any observable differences?